crates/core/src/blob.rs

```diff
@@ -66,17 +66,46 @@ impl<T: Default> Initialize<T> for BlobTypeMap<T> {
     }
 }
 
+new_id!(BlobId, "blob");
+
+pub trait PackedId: Copy + Into<BlobId> + From<BlobId> {
+    const TYPE: BlobType;
+}
+
+#[macro_export]
+/// Generate newtypes for `Id`s identifying packed blobs
+macro_rules! new_blobid {
```

Comment:
Would name it something like `impl_blob_id` to make it easier readable and make people aware, that this just adds `impl` to a type.

Reply:
renamed.